docs/gui/menus.md

```diff
@@ -0,0 +1,343 @@
+
+!!! note
+    Since `Container`s used by `Slot`s are deprecated by Forge in favor of using the [`IItemHandler` capability][cap], the rest of the explanation will revolve around using `SlotItemHandler`: the capability variant.
+
+A `SlotItemHandler` contains four parameters: the `IItemHandler` representing the inventory the stacks are within, the index of the stack this slot is specifically representing, and the x and y position of where the top-left position of the slot will render on the screen relative to `AbstractContainerScreen#leftPos` and `#topPos`. The client menu constructor should always supply an empty instance of an inventory of the same size.
+
+In most cases, any slots the menu contains is first added, followed by the player's inventory, and finally concluded with the player's hotbar. To access any individual `Slot` from the menu, the index must be calculated based upon the order of which slots were added.
+
+A `DataSlot` is an abstract class which should implement a getter and setter to reference the data stored in the data storage object.The client menu constructor should always supply a new instance via `DataSlot#standalone`.
+
+These, along with slots, should be recreated every time a new menu is initialized.
+
+!!! warning
+    Although a `DataSlot` may hold enough data for an integer, only a short is sent across the network.
+
+```java
+// Assume we have an inventory from a data object of size 5
+// Assume we have a DataSlot constructed on each initialization of the server menu
+
+// Client menu constructor
+public MyMenuAccess(int containerId, Inventory playerInventory) {
+  this(containerId, playerInventory, new ItemStackHandler(5), DataSlot.standalone());
+}
+
+// Server menu constructor
+public MyMenuAccess(int containerId, Inventory playerInventory, IItemHandler dataInventory, DataSlot dataSingle) {
+  // Check if the data inventory size is some fixed value
+  // Then, add slots for data inventory
+  this.addSlot(new SlotItemHandler(dataInventory, /*...*/));
+
+  // Add slots for player inventory
+  this.addSlot(new Slot(playerInventory, /*...*/));
+
+  // Add data slots for handled integers
+  this.addDataSlot(dataSingle);
+
+  // ...
+}
+```
+
+#### `ContainerData`
+
+If multiple integers need to be synced to the client, a `ContainerData` can be used to reference the integers instead. This interface functions as an index lookup such that each index represents a different integer. `ContainerData`s can also be constructed in the data object itself if the `ContainerData` is added to the menu through `#addDataSlots`. The method creates a new `DataSlot` for the amount of data specified by the interface. The client menu constructor should always supply a new instance via `SimpleContainerData`.
```

Comment:
These also fall to the same caveat as `DataSlot`s with regards to the effectively `short` data size, right?